include/boost/graph/maximum_weighted_matching.hpp

```diff
@@ -54,7 +54,7 @@ namespace boost
         {
             typedef boost::iterator_property_map<typename std::vector<T>::iterator, VertexIndexMap> type;
         };
-        typedef decltype(graph::detail::V_EVEN) vertex_state_t; // anonymous enum defined elsewhere
+        typedef unsigned short vertex_state_t;
```

Comment:
Does this type ever change in graph?

Reply:
It's defined as anonymous enum and used only in `graph/max_cardinality_matching.hpp`,
see: https://github.com/boostorg/graph/blob/develop/include/boost/graph/max_cardinality_matching.hpp#L29

Reply:
Then could it be removed entirely and not just its use here?

Reply:
"Here" is `graph/maximum_weighted_matching.hpp`, this enum is defined (and used) in `graph/max_cardinality_matching.hpp`, not by me. 
Another idea is to assign a name for this enum in `graph/max_cardinality_matching.hpp` but I think maybe it's better not to tamper with existed files

Reply:
I'm a little uncomfortable with changing something which seems to be related to a template parameter.  If this is just for testing time, it is probably best to leave it be.

Reply:
Did you mean leave `decltype(graph::detail::V_EVEN)` be? it won't compile by g++
I understand you concern, however `vertex_state_t` has only possible values of {0, 1, 2} (no chance to be changed by logic) and only used in `graph/max_cardinality_matching.hpp`, so unsigned short shouldn't cause any problem I think

Reply:
OK.  This is new code so good enough for me at this point.  I'll read through the PR in more detail when the tests pass.